pdi/src/pdi.cxx

```diff
@@ -143,6 +143,29 @@ void warn_status(PDI_status_t status, const char* message, void*)
 	}
 }
 
+/** A structure to reclaim the datas properly in case of error
+ */
+struct Var_to_reclaim: public std::list<string> {
```

Comment:
I really like the idea. This seems to go the way of what was proposed above. I would like it to be be public and used on the provider side rather than on the user side though.

Also, as a side note, inheriting from a `std::list` is not something I would do. I'd rather have it as a member. Cf. https://en.wikipedia.org/wiki/Composition_over_inheritance